### PR TITLE
Idaho child tax credit

### LINKED ISSUE #2292: Idaho child tax credit
Reference: https://legislature.idaho.gov/statutesrules/idstat/Title63/T63CH30/SECT63-3029L/

### PR DESCRIPTION
**Pull request by LeqiGogo**
Fixes #2292

### Summary
🧒📄🌎

This pull request adds a new parameter file for the `amount` of the Idaho child tax credit in `policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/`. This is part of the effort to model state-level tax and benefit policies in the policy engine US.

> _`Idaho` credit_
> _A new parameter file_
> _Autumn leaves fall fast_

### Walkthrough
*  Add a new parameter file for the Idaho child tax credit amount ([link](https://github.com/PolicyEngine/policyengine-us/pull/2293/files?diff=unified&w=0#diff-a84aeb84cc0b9a7d7a199fc7726a34f1dcd40bbe10081431e94feb4ef727f17dR1-R9))

### CODE REVIEW

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/eligible_age.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+description: Idaho limits its child exemption to children below this age.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/eligible_age.yaml:1]:
```suggestion
description: Idaho limits its child tax credit to children below this age.
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/eligible_age.yaml
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/eligible_age.yaml:1]: Remove this file and point to `ctc_qualifying_children` to reflect the Idaho legal code pointing to the IRC:
>For purposes of this section, the term "qualifying child" has the meaning as defined in section 24(c) of the Internal Revenue Code.

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+description: Idaho allows this amount for each child under the child tax credit.
+values:
+  2018-01-01: 205 
+metadata:
+  unit: currency-USD
+  label: Idaho child tax credit amount
+  reference:
+    - title: IDAHO STATUTES - TITLE 63 - 067 - SECTION 63-3029L
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/amount.yaml:8]: Add a tax form reference to check potential inflation adjustment

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+    def formula(tax_unit, period, parameters):
+        # Get relevant parameter subtree.
+        p = parameters(period).gov.states.id.tax.income.credits.child_credit
+        # Determine eligibility for each person in the tax unit.
+        person = tax_unit.members
+        meets_age_limit = person("age", period) < p.eligible_age
+        eligible = meets_age_limit & person("is_child_of_tax_head", period)
+        # Count number of eligible children in the tax unit.
+        count_eligible = tax_unit.sum(eligible)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py:20]:
```suggestion
        count_eligible = tax_unit("ctc_qualifying_children", period)
```

File: policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        # Get relevant parameter subtree.
+        p = parameters(period).gov.states.id.tax.income.credits.child_credit
+        # Determine eligibility for each person in the tax unit.
+        person = tax_unit.members
+        meets_age_limit = person("age", period) < p.eligible_age
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py:17]:
```suggestion
```

File: policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+        eligible = meets_age_limit & person("is_child_of_tax_head", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py:18]:
```suggestion
```

File: policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+    label = "Idaho Child Credit"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        # Get relevant parameter subtree.
+        p = parameters(period).gov.states.id.tax.income.credits.child_credit
+        # Determine eligibility for each person in the tax unit.
+        person = tax_unit.members
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py:16]:
```suggestion
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/eligible_age.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/eligible_age.yaml:1]: delete this file

File: policyengine_us/parameters/gov/states/id/tax/income/deductions/retirement benefit/age_eligibility.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/deductions/retirement benefit/age_eligibility.yaml:1]: these are retirement benefit deduction files, please delete

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+description: Idaho allows child tax credit for children below this age.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml:1]:
```suggestion
description: Idaho provides the child tax credit for children below this age.
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+values:
+  2018-01-01: 16 
+metadata:
+  unit: age
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml:5]:
```suggestion
  unit: year
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+  label: Idaho child tax credit age limit
+  reference:
+    - title: IDAHO STATUTES - TITLE 63 - 067 - SECTION 63-3029L
+      href: https://legislature.idaho.gov/statutesrules/idstat/Title63/T63CH30/SECT63-3029L/
+    - title: Form 40 2022 Individual Income Tax Return
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml:10]: use exact page number

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml [hunk 10]
```diff
@@ -0,0 +1,11 @@
+description: Idaho allows child tax credit for children below this age.
+values:
+  2018-01-01: 16 
+metadata:
+  unit: age
+  label: Idaho child tax credit age limit
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml:6]:
```suggestion
  label: Idaho child tax credit age threshold
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/amount.yaml [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+      href: https://legislature.idaho.gov/statutesrules/idstat/Title63/T63CH30/SECT63-3029L/
+    - title: Form 40 2022 Individual Income Tax Return
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/amount.yaml:10]: add page

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/amount.yaml [hunk 12]
```diff
@@ -0,0 +1,11 @@
+description: Idaho allows this amount for each child under the child tax credit.
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/amount.yaml:1]:
```suggestion
description: Idaho provides this child tax credit amount for each child.
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml (hunk 7 above)
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligiblity.yaml:1]:
```suggestion
description: Idaho provides the child tax credit for children below this age.
```

File: policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py [hunk 13]
```diff
@@ -0,0 +1,21 @@
...
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        # Get relevant parameter subtree.
+        p = parameters(period).gov.states.id.tax.income.credits.child_credit
+        # Count number of eligible children in the tax unit.
+        age = person("age", period)
+        eligible_child = age < p.age_eligibility
+        eligible_children = tax_unit.sum(eligible_child)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py:19]: the law points to the section 24(c) of the Internal Revenue Code - use ctc_qualifying_children.py as the number of children

eligible_children = tax_unit("ctc_qualifying_children", period)

File: policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py [hunk 14]
```diff
@@ -0,0 +1,21 @@
...
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        person = tax_unit.members
+        # Get relevant parameter subtree.
+        p = parameters(period).gov.states.id.tax.income.credits.child_credit
+        # Count number of eligible children in the tax unit.
+        age = person("age", period)
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py:17]:
```suggestion
```

File: policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py [hunk 15]
```diff
@@ -0,0 +1,21 @@
...
+        eligible_child = age < p.age_eligibility
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/credits/child_credit/id_child_credit.py:18]:
```suggestion
```

File: policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligibility.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/id/tax/income/credits/child_credit/age_eligibility.yaml:1]: delete this parameter - dont need

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/variables/gov/states/id/tax/income/credits/id_ctc.py [hunk 16]
```diff
@@ -0,0 +1,18 @@
...
+    entity = TaxUnit
+    label = "Idaho Child Tax Credit"
+    unit = USD
+    definition_period = YEAR
+    defined_for = StateCode.ID
+
+    def formula(tax_unit, period, parameters):
+        # Get relevant parameter subtree.
+        p = parameters(period).gov.states.id.tax.income.credits.ctc
+        # Count number of eligible children under the federal ctc.
```
Comment (MaxGhenis) [inline policyengine_us/variables/gov/states/id/tax/income/credits/id_ctc.py:15]:
```suggestion
        # "the term "qualifying child" has the meaning as defined in section 24(c) of the Internal Revenue Code."
        # IRC 24(c) defines eligible children under the federal CTC.
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**